blockchain/blockchain.go

```diff
@@ -100,12 +100,12 @@ type PersistedState struct {
 }
 
 func LoadOrNewBlockchain(db dbm.DB, genesisDoc *genesis.GenesisDoc,
-	logger *logging.Logger) (*blockchain, error) {
+	logger *logging.Logger) (MutableBlockchain, error) {
```

Comment:
it's not always correct but I think "accept interfaces, return structs" is a reasonable mantra here. An exception would be if we really can't trust the caller with a `*blockchain` - do you think this is the case here? Since `blockchain` is unexported it's really only for the benefit of this package - returning the struct makes it easier to test.

If anything I have been stepping back from my use of 'interfaces defined in same package as struct' in this case I think it is useful to define the `CommitBlock` function centrally but on reflection I'm questioning whether I need the `MutableBlockchain` at all...

Reply:
In case we need to have Immutable-Blockchain, it might helps. I think your idea is better. Better to not make it so complicated. I will revert this change.

Reply:
:+1: